BitFaster.Caching/IScopedCache.cs

```diff
@@ -28,6 +28,11 @@ public interface IScopedCache<K, V> where V : IDisposable
         /// </summary>
         ICacheMetrics Metrics { get; }
 
+        /// <summary>
+        /// Gets the cache events.
+        /// </summary>
+        ICacheEvents<K, Scoped<V>> Events { get; }
```

Comment:
This exposes the scope object to the outside world, enabling bad things to happen (e.g. caller disposes the scope, now the cache contains a disposed object).

Alternative would be to make some sort of wrapper to proxy the event and pull the object out, sending a potentially disposed object to the event handler.

It seems better to send the scope. The caller can then create a lifetime if they need to work on the value, else ignore it.

Reply:
Added comment.